sdk/core/Azure.Core.TestFramework/src/RetryOnErrorAttribute.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
+
+namespace Azure.Core.TestFramework
+{
+    /// <summary>
+    /// Attribute used to specify that a test must be retried in the specific case of a known error. If the test
+    /// continues to fail due to this particular issue after reaching the specified limit of tries, the test result is
+    /// marked as inconclusive.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    public class RetryOnErrorAttribute : NUnitAttribute, IRepeatTest
+    {
+        private readonly int _tryCount;
+
+        private readonly Func<TestExecutionContext, bool> _shouldRetry;
+
+        /// <summary>
+        /// Initializes a <see cref="RetryOnErrorAttribute"/> instance.
+        /// </summary>
+        /// <param name="tryCount">The number of times that the test can be executed.</param>
+        /// <param name="shouldRetry">The function that determines whether the test should be retried.</param>
+        public RetryOnErrorAttribute(int tryCount, Func<TestExecutionContext, bool> shouldRetry)
```

Comment:
This couldn't possibly be serialized as an attribute parameter, could it? If this is intended to be derived, I'd make the class abstract and this constructor protected. Or would the default implementation be good enough with just a `tryCount` to put in a public constructor, and still make this protected but keep the class instantiable?